pennylane_lightning/core/src/simulators/lightning_tensor/tensor/base/TensorBase.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file TensorBase.hpp
+ * Tensor base class for all backends.
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "Error.hpp"
+
+namespace Pennylane::LightningTensor {
+
+/**
+ * @brief CRTP-enabled base class for tensor.
+ *
+ * @tparam Precision Floating point precision.
+ * @tparam Derived Derived class to instantiate using CRTP.
+ */
+template <class PrecisionT, class Derived> class TensorBase {
```

Comment:
Could you please explain what `template this on Derived` means here?

Reply:
I mean, why bother creating CRTPs if we're not using CRTP features? I would avoid CRTPing where we can avoid it. Since the base class does not use `Derived`, I would just use regular inheritance. Same for the other class. This is unlike `StateVectorBase.hpp` which calls the function implementations of specialized versions of itself.

Reply:
As discussed, base class here is not complete yes and could use `Derived` in the future for other features.   I will keep current design.